src/diagonal.works/b6/ingest/basic.go

```diff
@@ -251,6 +257,39 @@ func (r relationFeature) ToGeoJSON() geojson.GeoJSON {
 	return b6.RelationFeatureToGeoJSON(r, r.features)
 }
 
+type collectionFeature struct {
+	*CollectionFeature
+	features b6.FeaturesByID
+}
+
+func newCollectionFeature(c *CollectionFeature, byID b6.FeaturesByID) collectionFeature {
+	return collectionFeature{c, byID}
+}
+
+func (c collectionFeature) ToGeoJSON() geojson.GeoJSON {
+	return b6.CollectionFeatureToGeoJSON(c, c.features)
+}
+
+func (c collectionFeature) Feature() b6.CollectionFeature {
+	return c
+}
+
+func (c collectionFeature) CollectionID() b6.CollectionID {
+	return c.CollectionFeature.CollectionID
+}
+
+func (c collectionFeature) Len() int {
+	return len(c.CollectionFeature.Keys)
```

Comment:
ok to assume num of values is same as keys ? i ll do "implements iterator interface" in follow up / too lazy now / but i think next() makes the same assumption

Reply:
ok actually read the code :) next really confused me for like 15min, i thought it was next? but turns out it was next!

done w collection interface; i had to do a reset / thing but actually i think that s ur begin from iterator.. i have a todo to implement references as well so i ll wrap those two guys up tmrw, if u re here before me and u need to feel free to merge

Reply:
ok yes that is exactly what begin does